zerver/management/commands/send_zulip_updates.py

```diff
@@ -0,0 +1,134 @@
+        previous_level = f"**Zulip updates level {level-1}**"
+        pattern = re.compile(
+            f"{re.escape(current_level)}(.*?)" + f"{re.escape(previous_level)}", re.DOTALL
+        )
+        match = re.search(pattern, content)
+        assert match is not None
+        return match.group(1).strip()
+
+
+class Command(ZulipBaseCommand):
+    help = """Script to send zulip updates to realms."""
+
+    @override
+    def handle(self, *args: Any, **options: str) -> None:
+        latest_zulip_updates_level = get_latest_zulip_updates_level()
+
+        realms = Realm.objects.filter(
+            Q(zulip_updates_level__isnull=True)
+            | Q(zulip_updates_level__lt=latest_zulip_updates_level),
+            deactivated=False,
+        )
+
+        for realm in realms:
+            realm_zulip_updates_level = realm.zulip_updates_level
+            sender = get_system_bot(settings.NOTIFICATION_BOT, realm.id)
+
+            messages = []
+            new_zulip_updates_level = None
+            with transaction.atomic():
+                if realm_zulip_updates_level is None:
+                    # realm predates the zulip updates feature.
+                    # DM the administrator to set or verify the stream for zulip updates.
+                    administrators = realm.get_human_admin_users()
+                    content = get_zulip_update_message_for_level(level=1)
+                    for administrator in administrators:
+                        direct_message = internal_prep_private_message(
+                            sender=sender, recipient_user=administrator, content=content
+                        )
+                        messages.append(direct_message)
+
+                    new_zulip_updates_level = 1
+
+                    RealmAuditLog.objects.create(
+                        realm=realm,
+                        event_type=RealmAuditLog.REALM_PROPERTY_CHANGED,
+                        event_time=timezone_now(),
+                        extra_data={
+                            RealmAuditLog.OLD_VALUE: None,
+                            RealmAuditLog.NEW_VALUE: 1,
+                            "property": "zulip_updates_level",
+                        },
+                    )
+                elif (
+                    realm_zulip_updates_level == 1
+                    and realm.zulip_update_announcements_stream is None
+                ):
+                    # We wait for a week after sending DMs to let admin configure stream
+                    # for zulip updates. After that, they miss updates until they don't configure.
+                    level_none_to_one_auditlog = RealmAuditLog.objects.filter(
+                        realm=realm,
+                        event_type=RealmAuditLog.REALM_PROPERTY_CHANGED,
+                        extra_data={
+                            RealmAuditLog.OLD_VALUE: None,
+                            RealmAuditLog.NEW_VALUE: 1,
```

Comment:
These values seem wrong, since we were already at 1?

Reply:
`zerver/lib/zulip_update_announcements.py` has the updated code with this logic inside `is_group_direct_message_sent_to_admins_atleast_one_week_ago`